### PR TITLE
enable keepalive pinging

### PR DESCRIPTION
**Pull request by asutula**
Signed-off-by: Aaron Sutula <hi@asutula.com>

### CODE REVIEW

File: core/core.go [hunk 1]
```diff
@@ -282,6 +283,7 @@ func NewTextile(ctx context.Context, conf Config) (*Textile, error) {
 		}),
 		grpcweb.WithAllowedRequestHeaders([]string{"Origin"}),
 		grpcweb.WithWebsockets(true),
+		grpcweb.WithWebsocketPingInterval(time.Second*5),
```
Comment (sanderpick) [inline core/core.go:286]: How about a package constant that users can tweak?

**Comment by carsonfarmer** [comment]
I hope you don't mind @asutula, but I added a configurable package constant to tweak this.

File: core/core.go (hunk 1 above)
Comment (carsonfarmer) [reply core/core.go]: Added

File: core/core.go [hunk 2]
```diff
@@ -63,6 +64,9 @@ var (
 	blockMethods = []string{
 		"/threads.pb.API/ListDBs",
 	}
+
+	// wsPingInterval controls the WebSocket keepalive pinging interval. Must be >= 1s.
+	wsPingInterval = time.Second * 5
```
Comment (sanderpick) [inline core/core.go:69]: If you make this exported, users can tweak it... an easy win

**Comment by asutula** [comment]
So, how does that variable get set if someone wants to change it @carsonfarmer? I guess there would some flags at the cli cmd level, that would filter down to set that var?

File: core/core.go (hunk 2 above)
Comment (carsonfarmer) [reply core/core.go:69]: 🤦 of course! good call

**Comment by carsonfarmer** [comment]
Based on my understanding, we could tweak it (now that its exported) from `bucks/main.go` and/or `hubs/main.go`, probs via some flags. Seems like we can leave it to the default for now, as I expect we're unlikely to want to change it? But I defs could be wrong and we could defs just add it as flags in there?

**Comment by asutula** [comment]
Ok, I'll merge it and we can address the flags thing potentially in another PR if we decide it's needed.

**Pull request merged**